packages/sdk/src/util.test.ts

```diff
@@ -439,25 +454,123 @@ export async function createSpaceAndDefaultChannel(
     }
 }
 
+export async function createVersionedSpaceFromMembership(
+    client: Client,
+    spaceDapp: ISpaceDapp,
+    wallet: ethers.Wallet,
+    name: string,
+    membership: LegacyMembershipStruct | MembershipStruct,
+): Promise<ethers.ContractTransaction> {
+    if (useLegacySpaces() && isLegacyMembershipType(membership)) {
+        return await spaceDapp.createLegacySpace(
+            {
+                spaceName: `${name}-space`,
+                uri: `${name}-space-metadata`,
+                channelName: 'general',
+                membership,
+            },
+            wallet,
+        )
+    } else {
+        if (isLegacyMembershipType(membership)) {
+            // Convert legacy space params to current space params
+            membership = {
+                settings: membership.settings,
+                permissions: membership.permissions,
+                requirements: {
+                    everyone: true,
+                    users: [],
+                    ruleData: encodeRuleDataV2(
+                        convertRuleDataV1ToV2(
+                            membership.requirements.ruleData as IRuleEntitlementBase.RuleDataStruct,
+                        ),
+                    ),
+                },
+            }
+        }
+        return await spaceDapp.createSpace(
+            {
+                spaceName: `${name}-space`,
+                uri: `${name}-space-metadata`,
+                channelName: 'general',
+                membership,
+            },
+            wallet,
+        )
+    }
+}
+
+// createVersionedSpace accepts either legacy or current space creation parameters and will
+// fall backto the legacy space creation endpoint on the spaceDapp if the appropriate flag is set.
+// If a user does not pass in a legacy space creation parameter, the function will not use
+// the legacy space creation endpoint, because the updated parameters are not backwards
+// compatible - we don't attempt conversion here.
+export async function createVersionedSpace(
+    spaceDapp: ISpaceDapp,
+    createSpaceParams: CreateSpaceParams | CreateLegacySpaceParams,
+    signer: SignerType,
+): Promise<ethers.ContractTransaction> {
+    if (useLegacySpaces() && isCreateLegacySpaceParams(createSpaceParams)) {
+        return await spaceDapp.createLegacySpace(createSpaceParams, signer)
+    } else {
+        if (isCreateLegacySpaceParams(createSpaceParams)) {
+            // Convert legacy space params to current space params
+            createSpaceParams = {
+                spaceName: createSpaceParams.spaceName,
+                uri: createSpaceParams.uri,
+                channelName: createSpaceParams.channelName,
+                membership: {
+                    settings: createSpaceParams.membership.settings,
+                    permissions: createSpaceParams.membership.permissions,
+                    requirements: {
+                        everyone: true,
```

Comment:
is `everyone` always true in V2? Or should this be the value of the `createSpaceParams.membership.requirements.everyone` ?

Reply:
it should!